### PR TITLE
[LinearGradient][iOS] fix implicit locations on iOS

### LINKED ISSUE #15695: [iOS] Linear gradient not working properly after upgrading to Expo SDK 44
### Summary

After updating to SDK 44, all the gradients presented in my application started to not work properly on iOS. In Android, everything works well.

After upgrading to SDK 44 (**gradient not working**):
SDK44_Not_Working

In SDK 43 **(everything OK**):
SDK43_Working

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

44

### Environment

  Expo CLI 5.0.1 environment info:
    System:
      OS: Windows 10 10.0.19043
    Binaries:
      Node: 14.17.4 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 6.14.14 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      babel-preset-expo: 9.0.1 => 9.0.1
      expo: ^44.0.0 => 44.0.1
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
      react-native-web: 0.17.1 => 0.17.1
    Expo Workflow: managed

### Reproducible demo

A snack with a simple code using expo-linear-gradient can be found here: https://snack.expo.dev/egmHJLklN.

However, as Expo Snack does not support Expo SDK 44 yet, it is not possible to see the error in the iOS emulator. The code needs to be placed locally in a project that uses SDK 44 to reproduce this.

Again, this only happens to iOS apps.

### PR DESCRIPTION
**Pull request by kbrandwijk**
# Why

- Resolves https://github.com/expo/expo/issues/15695

# How

The Swift rewrite introduced a default locations array of [0,1]. This caused an issue when >2 colors were specified, and no locations were provided, because the first two colors would always use 0 and 1 from the default locations array. I removed the default values, so that when no locations are specified, they are always calculated.

# Test Plan

Tested manually with a couple of different gradients, with and without locations, with 2 and more colors, all work well now.

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-linear-gradient/ios/LinearGradientLayer.swift [hunk 1]
```diff
@@ -5,7 +5,7 @@ import CoreGraphics
 
 var defaultStartPoint = CGPoint(x: 0.5, y: 0.0)
 var defaultEndPoint = CGPoint(x: 0.5, y: 1.0)
-var defaultLocations: [CGFloat] = [0.0, 1.0]
+var defaultLocations: [CGFloat] = []
```
Comment (bbarthec) [inline packages/expo-linear-gradient/ios/LinearGradientLayer.swift:8]: I can see that our docs are explicitly documenting this behaviour https://docs.expo.dev/versions/latest/sdk/linear-gradient/#lineargradientprops 🤔

**Review by bbarthec — CHANGES_REQUESTED** [review]
I see that there's discrepancy between the docs and proposed behaviour 🤔

File: packages/expo-linear-gradient/ios/LinearGradientLayer.swift (hunk 1 above)
Comment (kbrandwijk) [reply packages/expo-linear-gradient/ios/LinearGradientLayer.swift:8]: I dug a bit deeper, and also removed the default value from the type definition that I overlooked initially.

Comment (bbarthec) [reply packages/expo-linear-gradient/ios/LinearGradientLayer.swift:8]: Yeah, I've got nothing against removing it in favour of something different, e.g. empty array.
And what about other platforms? Web and Android? Is the new default value propagated there as well? 🤔 
We don't want to introduce any discrepancies between something that is consistent right now.

**Review by bbarthec — COMMENTED** [review]
As you've modified the JS/TS side you also need to rebuild the package (`yarn build`) and commit the newly generated files.

**Comment by kbrandwijk** [comment]
@bbarthec It's actually the other way around. The current implementation is _not_ consistent. The default value comes from the Swift rewrite for iOS, it is the _only_ platform that actually defines this default value, web (https://github.com/expo/expo/blob/master/packages/expo-linear-gradient/src/NativeLinearGradient.web.tsx#L52) and android (https://github.com/expo/expo/blob/master/packages/expo-linear-gradient/android/src/main/java/expo/modules/lineargradient/LinearGradientView.java#L18
) do not!

So this PR actually brings back the consistency that existed before the rewrite (no default on all platforms). I'm not introducing a breaking change on either of the three platforms either, it is just a bugfix for iOS, nothing more.

**Comment by kbrandwijk** [comment]
> As you've modified the JS/TS side you also need to rebuild the package (`yarn build`) and commit the newly generated files.

Sorry I missed that. One repo requires local build + commit, the other repo does it in a pipeline as part of the release process only. Hard to keep track sometimes ;)

**Review by bbarthec — APPROVED** [review]
Ok, thank you for clarifying the behaviour of each platform 👏 
Given that the iOS became faulty after Swift rewrite I agree that it's a fix then 💪 
And yeah, I agree that the different behaviour of each repository's commit/publish pipelines is causing a headache - I believe we'll try limit the manual steps in the nearest future in this repo too 😁

**Review by tsapeta — APPROVED** [review]
Good catch! Thanks for fixing my fault 🙈 
I think it might also be worth documenting what happens when the locations are not provided 🤔

**Comment by kbrandwijk** [comment]
> Good catch! Thanks for fixing my fault 🙈 I think it might also be worth documenting what happens when the locations are not provided 🤔

I was deliberating adding something to the docs like 'if ommitted, the colors will automatically be evenly distributed' or sth like that. Do you think that'll be helpful?

**Comment by tsapeta** [comment]
Yes, that would be helpful, thanks! Looks like you need to rebuild the files again now 😅

**Review by tsapeta — APPROVED** [review]

File: packages/expo-linear-gradient/CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,9 @@
 
 ### 🐛 Bug fixes
 
+- Prevent crashes by adding unimplemented `CALayer` initializer `init(layer:)`. ([#15843](https://github.com/expo/expo/pull/15843) by [@dillonhafer](https://github.com/dillonhafer))
```
Comment (EvanBacon) [inline packages/expo-linear-gradient/CHANGELOG.md:11]:
```suggestion
```

**Pull request merged**